### PR TITLE
add path helper functions to lightclient

### PR DESCRIPTION
**Pull request by fluidvanadium**

### CODE REVIEW

**Review by zancas — COMMENTED** [review]
This looks pretty reasonable, but if we're tweaking the pub interface to config, we need to check with our consumers, @AArnott and @juanky201271 (at least) to see their projects are affected.

File: zingocli/src/lib.rs [hunk 1]
```diff
@@ -415,7 +415,7 @@ pub fn startup(
             false,
         )?),
         None => {
-            if config.wallet_exists() {
+            if config.wallet_path_exists() {
```
Comment (zancas) [inline zingocli/src/lib.rs:418]: I like this.

File: zingolib/src/lightclient.rs [hunk 2]
```diff
@@ -2055,6 +2056,14 @@ impl LightClient {
 
         Ok(())
     }
+
+    pub fn get_wallet_file_location(&self) -> PathBuf {
```
Comment (zancas) [inline zingolib/src/lightclient.rs:2060]: Can you add  a doc-comment to this fn and start a CHANGELOG.md following the style here:

https://github.com/zcash/librustzcash/blob/main/zcash_primitives/CHANGELOG.md

File: zingoconfig/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## 1.3.2 2023-10
```
Comment (zancas) [inline zingoconfig/CHANGELOG.md:10]: What's this number?

File: zingolib/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## 1.3.2 2023-10
```
Comment (zancas) [inline zingolib/CHANGELOG.md:10]: What's this number?

Comment (zancas) [reply zingolib/CHANGELOG.md:10]: I think your changes are going into `Unreleased` so we can just delete this number, which I copied form the keepachangelog URL.

File: zingolib/src/lightclient.rs [hunk 5]
```diff
@@ -2055,6 +2057,27 @@ impl LightClient {
 
         Ok(())
     }
+
+    /// Some LightClients have a data dir in state. Mobile versions instead rely on a buffer and will return an error if this function is called.
```
Comment (zancas) [inline zingolib/src/lightclient.rs:2061]: I see the exact same doc-comment twice.

Here's an an example of a doc-comment in `librustzcash`:

https://github.com/zcash/librustzcash/blob/29c676ff125406e4b9f20fd046d103f42aa85e31/zcash_primitives/src/transaction/fees/zip317.rs#L42

Run `cargo doc --open` to see how the doc-comment is rendered.

File: zingolib/src/lib.rs [hunk 6]
```diff
@@ -6,6 +6,7 @@ mod test_framework;
 pub mod blaze;
 pub mod commands;
 pub mod compact_formats;
+pub mod error;
```
Comment (zancas) [inline zingolib/src/lib.rs:9]: @AloeareV @fluidvanadium why not make `error` into a crate?

`zingo-error`?

Comment (fluidvanadium) [reply zingolib/src/lib.rs:9]: each crate should have its own error enum

**Review by zancas — APPROVED** [review]

**Comment by AArnott** [comment]
I just upgraded. The obsolete message that pointed me to the new name was helpful.

**Pull request merged**